AzureMonitorAgent/agent.py

```diff
@@ -1192,34 +1194,112 @@ def generate_localsyslog_configs():
     """
     Install local syslog configuration files if not present and restart syslog
     """
-    if os.path.exists('/etc/rsyslog.d/') and not os.path.exists('/etc/rsyslog.d/10-azuremonitoragent.conf'):
-        copyfile("/etc/opt/microsoft/azuremonitoragent/syslog/rsyslogconf/05-azuremonitoragent-loadomuxsock.conf","/etc/rsyslog.d/05-azuremonitoragent-loadomuxsock.conf")
-        copyfile("/etc/opt/microsoft/azuremonitoragent/syslog/rsyslogconf/10-azuremonitoragent.conf","/etc/rsyslog.d/10-azuremonitoragent.conf")
-        os.chmod('/etc/rsyslog.d/05-azuremonitoragent-loadomuxsock.conf', stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IROTH)
-        os.chmod('/etc/rsyslog.d/10-azuremonitoragent.conf', stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IROTH)
-        run_command_and_log(get_service_command("rsyslog", "restart"))
-        hutil_log_info("Installed local syslog configuration files and restarted syslog")
 
-    if os.path.exists('/etc/syslog-ng/syslog-ng.conf') and not os.path.exists('/etc/syslog-ng/conf.d/azuremonitoragent.conf'):
-        syslog_ng_confpath = os.path.join('/etc/syslog-ng/', 'conf.d')
-        os.makedir(syslog_ng_confpath)
-        copyfile("/etc/opt/microsoft/azuremonitoragent/syslog/syslog-ngconf/azuremonitoragent.conf","/etc/syslog-ng/conf.d/azuremonitoragent.conf")
-        os.chmod('/etc/syslog-ng/conf.d/azuremonitoragent.conf', stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IROTH)
-        run_command_and_log(get_service_command("syslog-ng", "restart"))
-        hutil_log_info("Installed local syslog configuration files and restarted syslog")
+    public_settings, _ = get_settings()
+    syslog_port = ''
+    if os.path.isfile(AMASyslogPortFilePath):
+        f = open(AMASyslogPortFilePath, "r")
+        syslog_port = f.read()
+        f.close()
+
+    if public_settings is not None and "useOmfwd" in public_settings and syslog_port != '':
```

Comment:
We should be very careful with publicly facing settings / contracts (e.g. "useOmfwd") as these will be set in stone once shipped or documented (and changing / removing them is a breaking change). Have we agreed on naming internally with PM?

Reply:
I am not sure if we want to document this as we only want to keep it behind a flag only for a release or two but I will ask PM team. I have opened another PR to fix the name: https://github.com/Azure/azure-linux-extensions/pull/1720